### PR TITLE
Add rows for all challenges

### PR DESCRIPTION
**Pull request by glokta1**
* Remove Docker challenge
* Add rows for all challenges 
* Update `data.yaml` with latest versions
* Update symbols for status to be more intuitive

### CODE REVIEW

**Comment by rohitpaulk** [comment]
Output looks good to me! @ryan-gang mind taking a look at this? You've got more context around the code here

**Comment by ryan-gang** [comment]
Hey @glokta1 can you please run `make lint` ?
The diff would be easier to parse that way. (Will integrate that into our CI at some point.)

**Comment by glokta1** [comment]
@ryan-gang Oh yup, sorry. Should be good now

File: data.yaml [hunk 1]
```diff
@@ -1,31 +1,31 @@
 c:
-  cycle: "17.0"
-  releaseDate: "2018-06-01"
+  cycle: "9.5"
+  releaseDate: "2022-05-27"
```
Comment (ryan-gang) [inline data.yaml:3]: From where did you get this release data @glokta1 ?
I was using [wikipedia](https://en.wikipedia.org/wiki/C_(programming_language)) for this btw, we want to be completely sure about this release cycle data.

**Comment by ryan-gang** [comment]
The `Language` enum order, decides the language ordering in the rendered README btw. Is the ordering as per expectations ?

**Comment by ryan-gang** [comment]
I was trying to keep our codebase strongly typed (as much as possible). `VersionedItem`'s type is not currently set up to store `None`. 
If it is **not too much effort,** can you try updating the types as such ? (Adding `Optional` as required, so as to pass None too)
mypy will complain if something is out of line, `make lint` should do it. 

(This is not very urgent tho, if something doesn't fit we can skip it, don't fret too much over it.)

**Review by ryan-gang — APPROVED** [review]
Looks good to me overall!

**Comment by glokta1** [comment]
> The `Language` enum order, decides the language ordering in the rendered README btw. Is the ordering as per expectations ?

Yes, that is intentional. Paul wanted me to follow this order: Rust, Go, Python, followed by other languages in alphabetical order.

Updated the types! @ryan-gang. `make lint` didn't complain now.

**Review by ryan-gang — APPROVED** [review]
LGTM!

File: data.yaml (hunk 1 above)
Comment (ryan-gang) [reply data.yaml:3]: Tagging you again, @glokta1 in case you missed this.

Comment (glokta1) [reply data.yaml:3]: @ryan-gang In naming the Dockerfiles  for C, we're using the GCC compiler version (`c-9.2.Dockerfile`) so I thought to make it consistent here. Used this: https://gcc.gnu.org/gcc-9/

This is not ideal of course. We should do the same thing we're doing for C++ where we mention the standard, not the associated compiler version. 

cc: @rohitpaulk

Comment (rohitpaulk) [reply data.yaml:3]: Ah. Yep we should change this, but wouldn't be backward-compatible - let's add to Linear to think & figure out how to best do this. For the current usage here, let's stick to whatever makes most sense in the meantime

Comment (glokta1) [reply data.yaml:3]: @rohitpaulk Created a Linear issue here: https://linear.app/codecrafters/issue/CC-1314/update-naming-of-c-dockerfile

Merging this as is for now.

**Pull request merged**